libs/pageserver_api/src/shard.rs

```diff
@@ -713,6 +713,27 @@ fn key_to_shard_number(count: ShardCount, stripe_size: ShardStripeSize, key: &Ke
     ShardNumber((hash % count.0 as u32) as u8)
 }
 
+/// For debugging, while not exposing the internals.
+#[derive(Debug)]
+#[allow(unused)] // used by debug formatting by pagectl
+struct KeyShardingInfo {
+    shard0: bool,
+    shard_number: ShardNumber,
+}
+
+pub fn describe(
+    key: &Key,
+    shard_count: Option<ShardCount>,
+    stripe_size: Option<ShardStripeSize>,
+) -> impl std::fmt::Debug {
+    let shard_count = shard_count.unwrap_or(ShardCount(8));
```

Comment:
For the sharding, I'd suggest not putting any default in (maybe omit the shard output if the user doesn't provide shard parameters) -- if we default to ShardCount(8) it will be confusing for other shard counts.

Reply:
Sounds good. This stuck as I was originally thinking the problem may have been sharding related.